src/derivatives/linalg/arithmetic.jl

```diff
@@ -270,7 +270,13 @@ end
 # a * b
 
 function reverse_mul!(output, output_deriv, a, b, a_tmp, b_tmp)
-    istracked(a) && increment_deriv!(a, mul!(a_tmp, output_deriv, transpose(value(b))))
+    if istracked(a)
+        if a_tmp isa AbstractVector && b isa AbstractMatrix
+            increment_deriv!(a, mul!(reshape(a_tmp, :, 1), output_deriv, transpose(value(b))))
```

Comment:
It seems this branch still causes test failures on some Julia versions - does it call a `mul!` method that was added only recently?

Reply:
It's not this line, it's line 288 below, where the same error occurs and requires the same fix.

Reply:
Ah sorry, I misread the error message then.